eth-contracts/contracts/TokenLocker.sol

```diff
@@ -27,69 +36,64 @@ contract TokenLocker {
         address indexed token,
         address indexed sender,
         uint256 amount,
-        string accountId
+        string  ckbAddress,
+        string replayResistOutpoint
     );
 
     event Unlocked(
         address indexed token,
-        uint128 amount,
-        address recipient
+        address indexed recipient,
+        uint256 amount
     );
 
-    // Function output from burning fungible token on Near side.
-    struct BurnResult {
-        uint128 amount;
-        address token;
-        address recipient;
-    }
-
-    constructor(address ckbSpvAddress, uint64 numConfirmations) public {
+    constructor(address ckbSpvAddress, uint64 numConfirmations, bytes32 typescriptCodeHash, uint8 typescriptHashType) public {
         ckbSpv_ = ICKBSpv(ckbSpvAddress);
         numConfirmations_ = numConfirmations;
-    }
-
-    function _decodeBurnResult(bytes memory txInfo) internal pure returns (BurnResult memory result) {
-        // TODO
-        // 1. check if txInfo matches  ckbTxProof.txHash
-        // 2. _decodeBurnResult from txInfo
-        uint128 mockAmount = 111100000000000000;
-        address mockToken = address(0);
-        address mockRecipient = address(0);
-        result = BurnResult(mockAmount, mockToken, mockRecipient);
+        recipientCellTypescriptCodeHash_ = typescriptCodeHash;
+        recipientCellTypescriptHashType_ = typescriptHashType;
     }
 
     // before lockToken, user should approve -> TokenLocker Contract with 0xffffff token
-    function lockToken(address token, uint256 amount, string memory ckbAddress) public {
+    function lockToken(address token, uint256 amount, string memory ckbAddress, string memory replayResistOutpoint) public {
```

Comment:
```suggestion
    function lockToken(address token, uint256 amount, bytes memory ckbAddress, bytes memory replayResistOutpoint) public {
```

Is bytes better for this? If we use string, we may have to hex encode it, the memory length will be longer, which may consume more gas.

Reply:
yes. Rename `ckbAddress` to `recipientLockscript`, and add `sudtExtraData` field